File: doctor_admin/app/Models/Fiche.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Fiche extends Model
{
    use HasFactory;
    protected $fillable = ['maladies_id', 'traitements', 'prevention', 'autres'];
    public function maladies()
    {
    return $this->belongsTo(Maladie::class);
    }
}

File: principal/lib/Screens/Models/fiche.dart
import 'dart:async';
import 'dart:convert';
import 'package:http/http.dart' as http;

class Fiche {
  final int id;
  final int maladies_id;
  final String traitements;
  final String prevention;
  final String autres;
  Fiche(
      {required this.id,
      required this.maladies_id,
      required this.traitements,
      required this.prevention,
      required this.autres});
  factory Fiche.fromJson(Map<String, dynamic> json) {
    return Fiche(
      id: json['id'] ?? 1,
      maladies_id: json['maladies_id'] ?? 1,
      traitements: json['traitements'] ?? '',
      prevention: json['prevention'] ?? '',
      autres: json['autres'] ?? '',
    );
  }
  
}
